cdap-docs/developer-manual/source/building-blocks/schedules.rst

```diff
@@ -8,169 +8,246 @@
-    scheduleProperties.put("anotherProperty", "anotherValue");
-
-    scheduleWorkflow(Schedules.builder("1MBStreamSchedule")
-                       .setDescription("Schedule triggered every 1MB of ingested data")
-                       .createDataSchedule(Schedules.Source.STREAM, "purchaseStream", 1),
-                     "MyWorkflow", scheduleProperties);
-    ...
-
-Every time the ``1MBStreamSchedule`` triggers, it passes the ``scheduleProperties`` as runtime arguments to the ``MyWorkflow``.
-
-The stream-size schedule uses two sources of information to know how much data has been ingested by a Stream.
-One is :ref:`stream size notifications <streams-notifications>`; the other is the aggregated metric that
-records the total size of events ingested by a stream.
-Notifications are only used as an indication that data is available. Stream-size schedules always confirm the
-information by querying the metric system, which is the canonical source of information.
-
-
-.. rubric:: Stream-size Schedules Lifecycle
-
-A stream-size schedule will execute a workflow every time the stream it references ingests an increment of data,
-also defined in the
-`schedule <../../reference-manual/javadocs/co/cask/cdap/api/schedule/Schedules.Builder.html#createDataSchedule(co.cask.cdap.api.schedule.Schedules.Source,%20java.lang.String,%20int)>`__.
+Run Constraints
+===============
 
-When a stream-size schedule is first created, during the deployment of an application, it will wait for the
-increment of data that it defined, starting from the current size of the stream as given by the Metric system.
+A run constraint can either delay or prevent the execution of a schedule’s workflow, based
+on a condition represented by the constraint. Whether the execution is delayed or canceled
+can be made explicit by specifying either ``.waitUntilMet()`` or ``abortIfNotMet()``
+when adding the constraint to the schedule builder. Every individual type of run constraint
+also has its own default for this behavior. These constraints are available:
+
+- ``withConcurrency(int n)``: Fulfilled if less than n runs of the same workflow are
+  currently executing. This is useful to limit the frequency and resource utilization
+  of a single workflow. By default, this aborts the job if not fulfilled.
+- ``withDelay(long n, TimeUnit unit)``: Fulfilled at least n time units after the job
+  is created. This is useful to delay the execution of a workflow after its trigger
+  fires, for example, if it is known that after some new data arrives, more new data
+  may arrive within short time, and the workflow should wait for that.
+- ``withTimeWindow(String startTime, String endTime)``: Fulfilled only in the time
+  window between the given start and end time. Both times are given in “HH:mm” form,
+  and an optional timezone can be given to interpret these times. By default, this
+  delays the execution of the job, but it can be configured to abort the job
```

Comment:
oh I see. I was expecting it to wait if all constraints are not met or abort if all constraints are not met, instead of having abort or wait for each individual constraint